### ISSUE TITLE
[RFE] RancherUI may allow to create StorageClass with provisioner harvester

### ISSUE THREAD

**Issue by Vicente-Cheng** [rancher/dashboard#7850]
**Is your feature request related to a problem? Please describe.**
Related to https://github.com/harvester/harvester/issues/3052

**Describe the solution you'd like**
After the various StorageClass supported on the harvester cluster. The guest cluster that is provisioned by the harvester should be allowed to create StorageClass with provisioner `driver.harvesterhci.io` and given the parameter setting with `hostStorageClass`.

**Describe alternatives you've considered**

**Additional context**
For now, we could create CtorageClass by importing the related manifest (described as below).
```yaml
allowVolumeExpansion: true
  apiVersion: storage.k8s.io/v1
  kind: StorageClass
  metadata:
    name: <StorageClass Name>
  provisioner: driver.harvesterhci.io
  reclaimPolicy: Delete
  volumeBindingMode: Immediate
  parameters:
    hostStorageClass: <StorageClass Name on the Host Harvester Cluster>
```

In the following picture, we could not choose the harvester provisioner.
<img width="1262" alt="storageclass_provisioner" src="https://user-images.githubusercontent.com/1615081/210690737-e2def3b8-db7a-48dc-94b7-19eb3c9edb13.png">

cc @bk201, @guangbochen

**Comment by mantis-toboggan-md** [comment]
Covered by https://github.com/rancher/dashboard/pull/7869

**Comment by TachunLin** [comment]
Verified **fixed** on `v2.7-head` and `v2.6-head`. Close this issue.

### Result
We can create Harveter storage class in guest cluster `create storage` page on Rancher UI 
 
Test plan 1: Rancher v2.7-head (1dd2af2), create RKE2 v1.25.9 guest cluster, use CSI driver 0.1.15 
Test plan 2: Rancher v2.6-head (4db94f6), create RKE2 v1.24.9 guest cluster, use CSI driver 0.1.11

* We can create storage class in guest cluster and select Harvester as provisioner 
  - v2.7-head 
  ![image](https://user-images.githubusercontent.com/29251855/219292732-fd48c72c-6f46-4787-b79a-0af668ff898d.png)
  - v2.6-head
  ![image](https://user-images.githubusercontent.com/29251855/219310858-75a8fea8-ca07-43c0-8ac3-7c9a8854220f.png)

* We can create Harvester PVC
  - v2.7-head
  ![image](https://user-images.githubusercontent.com/29251855/219293095-1eda3526-8b8b-4079-9fe6-44c4e2876c67.png)
  - v2.6-head 
  ![image](https://user-images.githubusercontent.com/29251855/219311067-847d7a7a-dbdd-43bd-98db-25aea115f0d4.png)
  
* Corresponding PVC can be created automatically
  - v2.7-head
  ![image](https://user-images.githubusercontent.com/29251855/219293204-862ae2c0-4656-4e1b-b71d-458d213f874b.png)
  - v2.6-head
  ![image](https://user-images.githubusercontent.com/29251855/219311210-924110cc-8cdc-4776-ad1f-f42f67b13231.png)

* Check the volumes page in Harvester dashboard, the PVC can be created 
  - v2.7-head
  ![image](https://user-images.githubusercontent.com/29251855/219293425-33f021d9-a793-481b-8f3f-c97448f93c6c.png)
  - v2.6-head
  ![image](https://user-images.githubusercontent.com/29251855/219311343-5993210f-46f8-43a1-9998-9e350e2d28c9.png)

### Test Information
* Test Environment: 1 node Harvester on local kvm machine
* Harvester version: master-213d4af7-head (02/14)
* Rancher version: v2.7-head , v2.6-head 
* RKE2 version: v1.25.9 (1dd2af2), v1.24.9 (4db94f6)

### Verify Steps
1. Import Harvester to Rancher and create cloud credential 
2. Provision an RKE2 guest cluster 
3. Open guest cluster management page 
4. Open Storage page to create a storage class
5. Select Harvester as the provisioner
6. Create a PVC based on the storage class 
7. Check the PV state in Rancher and Harvester
